openedx/core/lib/xblock_serializer/block_serializer.py

```diff
@@ -37,6 +37,20 @@ def __init__(self, block):
         # Search the OLX for references to files stored in the course's
         # "Files & Uploads" (contentstore):
         self.olx_str = utils.rewrite_absolute_static_urls(self.olx_str, course_key)
+
+        # If a block supports explicit assets, there's no need to scan the
+        # content looking for static assets that it *might* be using. Learning
+        # Core backed content supports this, which currently means v2 Content
+        # Libraries.
+        runtime_supports_explicit_assets = hasattr(block.runtime, 'get_block_assets')
+        if runtime_supports_explicit_assets:
+            self.static_files.extend(
+                block.runtime.get_block_assets(block)
+            )
+            return
```

Comment:
Nit: I'd recommend moving this code and the code below into a new `_find_static_assets` method or something like that. Or use an explicit `else:` case. I'm just worried  that someone might not notice the conditional early `return`  here, and add some unrelated code to the end of the constructor, and later encounter a bug when it sometimes doesn't run.